cou/apps/app.py

```diff
@@ -640,3 +640,46 @@ def _get_reached_expected_target_plan(
             function=self._check_upgrade,
             target=target,
         )
+
+
+@AppFactory.register_application(["designate-bind", "gnocchi"])
+class OpenStackChannelBasedApplication(OpenStackApplication):
```

Comment:
would it be possible to modify the base openstack app to handle these special cases? For example by falling back to `self.channel_codename` if workload version is not available.